### ISSUE TITLE
[Release process] Block release candidate creation if release notes are not present

### ISSUE THREAD

**Issue by gaiksaya** [opensearch-project/opensearch-build#2658]
### Is your feature request related to a problem? Please describe

See https://github.com/opensearch-project/opensearch-build/issues/2620 for more details. 

The release notes are something that is delaying a release, causing the commit id to differ from what was shipped to customer and are essential to give a great customer experience for the product.

Today we only have the [mechanism](https://github.com/opensearch-project/opensearch-build/tree/main/jenkins/release-notes-check) to check if the release notes are present or absent for a component. However, there is no way to enforce it.

### Describe the solution you'd like

Let the build pipeline fail if the release notes are not present for a component. Have a boolean flag that will enabled when we start the release process. For example, enable the flag (set to true) when code cut off date is near. 
Similarly we can disable it for `patch` releases where not all plugins need to have the release notes. 

### Describe alternatives you've considered

Manual ask the component teams to add release notes

**Comment by dblock** [comment]
Will a campaign falling out of https://github.com/opensearch-project/OpenSearch/issues/1868 make this unnecessary? Maybe that's worth the effort?

cc: @kotwanikunal

**Comment by kotwanikunal** [comment]
> Will a campaign falling out of [opensearch-project/OpenSearch#1868](https://github.com/opensearch-project/OpenSearch/issues/1868) make this unnecessary? Maybe that's worth the effort?
> 
> cc: @kotwanikunal

Yes. I am trying to get this initial solution to the point where releasing the notes becomes a GHA/workflow that can auto publish the notes.

**Comment by peterzhuamazon** [comment]
[Triage] @prudhvigodithi could you add more details on how to proceed with this?
Such as preventing the build if there is no release notes for all components.
Thanks.

**Comment by gaiksaya** [comment]
We have added release notes as one of the entry criteria in new release window model. https://github.com/opensearch-project/.github/blob/main/RELEASING.md#entrance-criteria-to-start-release-window

We also added a release-notes checker that would check for release notes in all repositories. Example: https://github.com/opensearch-project/opensearch-build/issues/3743#issuecomment-1712176296

Closing this issue!
